plasmapy/particles/particle_class.py

```diff
@@ -1685,6 +1723,27 @@ def __repr__(self):
         """
         return f"DimensionlessParticle(mass={self.mass}, charge={self.charge})"
 
+    def particle_dict(self) -> dict:
+        """
+        Return a dictionary representation of the minimum description of a dimensionless particle.
+
+        Examples
+        --------
+        >>> from plasmapy.particles import DimensionlessParticle
+        >>> dimensionless_particle = DimensionlessParticle(mass=1.0, charge=-1.0)
+        >>> dimensionless_particle.particle_dict()
+        {'plasmapy_particle': {'type': 'DimensionlessParticle', 'description': {'mass': '1.0', 'charge': '-1.0'}}}
+        >>> dimensionless_particle = DimensionlessParticle(mass=1.0)
+        >>> dimensionless_particle.particle_dict()
+        {'plasmapy_particle': {'type': 'DimensionlessParticle', 'description': {'mass': '1.0', 'charge': 'nan'}}}
+        """
+        particle_dictionary = super().particle_dict
+        particle_dictionary["plasmapy_particle"]["description"] = {
+            "mass": str(self.mass),
+            "charge": str(self.charge)
+        }
+        return particle_dictionary
+
```

Comment:
```suggestion
```

This is all part of the default behavior for a particle class, so it should be apart of `AbstractParticle` and not required to be overridden by its sub-classes.  Sub-classes should only override this if they are doing something abnormal, like the `Particle` class.  `Particle` has a different `__init__` structure than all the other sub-classes, that's why it's the alternative case in my [example](https://github.com/PlasmaPy/PlasmaPy/pull/836#issuecomment-640249264).

Reply:
I see. This makes sense. I think I had misunderstood the previous comment - sorry about that. I agree with overriding for the `Particle` instead of for `CustomParticle` and `DimensionlessParticle`. I still have the question about `Particle` throwing an exception when we try and get it's charge for something like an elemental particle.